src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -1149,6 +1149,25 @@ private static void PartialMethodChecks(SourceOrdinaryMethodSymbol definition, S
         {
             Debug.Assert(!ReferenceEquals(definition, implementation));
 
+            MethodSymbol constructedDefinition = definition;
+            if (implementation.IsGenericMethod)
+            {
+                constructedDefinition = definition.Construct(implementation.TypeArgumentsWithAnnotations);
+            }
+
+            bool returnTypesEqual = constructedDefinition.ReturnTypeWithAnnotations.Equals(implementation.ReturnTypeWithAnnotations, TypeCompareKind.AllIgnoreOptions);
+            if (!returnTypesEqual
+                && !SourceMemberContainerTypeSymbol.IsOrContainsErrorType(implementation.ReturnType)
+                && !SourceMemberContainerTypeSymbol.IsOrContainsErrorType(definition.ReturnType))
+            {
+                diagnostics.Add(ErrorCode.ERR_PartialMethodReturnTypeDifference, implementation.Locations[0]);
```

Comment:
>ERR_PartialMethodReturnTypeDifference [](start = 42, length = 37)

Should these errors be reported in `MergePartialMembers()`, where `ErrorCode.ERR_PartialMethodInconsistentTupleNames` is reported, where we have constructed methods?